### ISSUE TITLE
logging and validation

### ISSUE THREAD

**Issue by yuchenhou** [yuchenhou/elephant#21]
![overfitting](https://cloud.githubusercontent.com/assets/8304443/13647397/d06ef038-e5e8-11e5-8af9-f0f2b4621472.png)
I managed to use the logging facility and I attached a screenshot below for a training log(training loss vs training steps). Currently skflow's dosen't have an efficient fit() method with validation (tensorflow/skflow#133) - fit() validates once per step instead of once per epoch, dramatically slowing down the fit() process. I'd like to improve this fit() process as it can benefit other users as well. We can expect this improvement to give us more efficient validation, over-fitting detection and early stopping.

**Comment by yuchenhou** [comment]
![movie](https://cloud.githubusercontent.com/assets/8304443/13713646/e56c1eb0-e77e-11e5-923c-4da7a6180788.png)
I updated the testing program to evaluate training error and validation error once per epoch. The model can decrease training error very well but not validation error after 2 epochs.

**Comment by lbholder** [comment]
Looks like validation error reaches its minimum at about 4 epochs, which seems rather soon, but will depend on the size of the dataset. Perhaps a slower learning rate would help.
